querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java

```diff
@@ -115,41 +115,28 @@ public List<Path<?>> getConstantPaths() {
      */
     @SuppressWarnings("unchecked")
     protected List<Expression<?>> getIdentifierColumns(List<JoinExpression> joins, boolean alias) {
-        if (joins.size() == 1) {
-            JoinExpression join = joins.get(0);
-            if (join.getTarget() instanceof RelationalPath) {
-                return ((RelationalPath) join.getTarget()).getColumns();
```

Comment:
This optimisation is gone.

Reply:
I didn't realize it was an optimization.